packages/service/ni_measurement_plugin_sdk_service/_internal/parameter/serialization_descriptors.py

```diff
@@ -22,15 +23,19 @@ def _create_enum_type_class(
 ) -> None:
     """Implement a enum class in 'file_descriptor'."""
     enum_dict = loads(parameter_metadata.annotations[ENUM_VALUES_KEY])
-    enum_type_name = _get_enum_type(parameter_metadata).__name__
 
-    # if enum is a protobuf then enum_type_name is 1st letter of each enum name
-    # e.g. {"NONE": 0, "RED": 1, "GREEN": 2} -> NRG
-    if enum_type_name == "int" or enum_type_name == "NoneType":
-        enum_field_names = list(enum_dict.keys())[:]
-        enum_type_name = "".join(name[0] for name in enum_field_names)
+    # True if enum is protobuf
+    if not type(parameter_metadata.enum_type) is type(Enum):
```

Comment:
Why not do the same check as the other file?

```suggestion
    if hasattr(parameter_metadata.enum_type, "ValueType"):
```

Reply:
You want me to change the try and except clause under that to the hasattr() too? Or is it best to keep as is?

Reply:
It looks like you are checking for the same condition: "is this a protobuf enum or a Python enum?" Is that right?

The best way to do the same thing in two places is to share the code. The second best way is to use the same logic. Using different logic is not as good because there may be cases where the two approaches have different results.

So the best way is to add a function that does the hasattr check and call it in both places.

Reply:
Earlier you commented this:

> For the case where we really don't know the enum type name, can we derive it from the parameter name? Converting the parameter name to PascalCase would be a good naming scheme.
> 
> You can convert snake_case to PascalCase by splitting on "_" and capitalizing each section.

Which is why there's the try and except to handle the case where enum_type.DESCRIPTOR.name isn't defined or do we not need it? I didn't ask why we need it since I didn't really think about it but is there a situation where the name isn't defined? Since from what I know, the name is required to create the enum type for a protobuf enum, right?